cellprofiler/modules/measurecorrelation.py

```diff
@@ -305,9 +517,163 @@ def run_image_pair_objects(self, workspace, first_image_name,
                                  labels, lrange))
             # Explicitly set the correlation to NaN for masked objects
             corr[scind.sum(1, labels, lrange) == 0] = np.NaN
+            result += [[first_image_name, second_image_name, object_name,"Mean Correlation coeff","%.3f"%np.mean(corr)],
+                       [first_image_name, second_image_name, object_name,"Median Correlation coeff","%.3f"%np.median(corr)],
+                       [first_image_name, second_image_name, object_name,"Min Correlation coeff","%.3f"%np.min(corr)],
+                       [first_image_name, second_image_name, object_name,"Max Correlation coeff","%.3f"%np.max(corr)]]
+
+            # Threshold as percentage of maximum intensity of objects in each channel
+            tff = (self.thr.value/100) * fix(scind.maximum(first_pixels,labels, lrange))
+            tss = (self.thr.value/100) * fix(scind.maximum(second_pixels,labels, lrange))
+        
```

Comment:
There's a problem in scipy.ndimage.sum if there are no values for a label:

    > scind.sum(np.zeros(0), np.zeros(0, int), np.arange(1, 2))
    Traceback (most recent call last):
      Python Shell, prompt 8, line 1
      File "C:\Python27\Lib\site-packages\scipy\ndimage\measurements.py", line 580, in sum
        count, sum = _stats(input, labels, index)
      File "C:\Python27\Lib\site-packages\scipy\ndimage\measurements.py", line 499, in _stats
         labels.min() < 0 or labels.max() > labels.size):
      File "C:\Python27\Lib\site-packages\numpy\core\_methods.py", line 21, in _amin
        out=out, keepdims=keepdims)
    ValueError: zero-size array to reduction operation minimum which has no identity

I am going to fix this so that you sum over the values * the truth array. That will give you values of zero for masked pixels which is the same as leaving them out.